Remove feature flag for editing custom fields in products and orders, and background tasks

joshheald opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses. -->
<!-- Part of: # -> use this if your PR is one of many related to one issue -->

## Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This PR removes the flag which was used for editing custom fields (metadata) in products and orders. Those features shipped a long time ago, so there's no need for the flag any more.

It also removes an unused `backgroundTasks` feature flag – I think Ernesto had removed all the sites it was called, but not the flag itself.

## Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
Check that you can still edit custom fields on products and orders.

Note that you edit order custom fields in Order Details, not by editing the overall order

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

joshheald committed:
Remove unused backgroundTasks feature flag

This flag wasn’t enabled, but it’s also not used anywhere in the code, so we can just remove it.

joshheald committed:
Remove viewEditCustomFieldsInProductsAndOrders flag

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

joshheald commented:
I'll fix the tests on these today

joshheald committed:
Update remaining tests for removed feature flag

joshheald committed:
Fix lint

joshheald committed:
Dark mode: corners and background when loading

Previously, the loading rows didn’t match the product rows, especially in dark mode.

This commit ensures that the rounded corners are shown, and a background shown in both modes.

joshheald committed:
Revert "Dark mode: corners and background when loading"

This reverts commit 250b1e2d341170fee58d7bf177a68533fefb6106.

staskus reviewed (approved):
Thanks for this ongoing feature flag clean-up 👍
